unixfs/src/walk.rs

```diff
@@ -511,20 +517,16 @@ impl InnerEntry {
     }
 
     fn set_path(&mut self, name: &str, depth: usize) {
-        debug_assert_eq!(self.depth, self.path.ancestors().count());
-
         while self.depth >= depth && self.depth > 0 {
-            assert!(self.path.pop());
+            assert!(path_pop(&mut self.path));
```

Comment:
This is a poor man's PathBuf::pop. If not good enough, it is possible to port std's PathBuf::(ancestors|pop) to a internal module for private usage.

Reply:
Started a new thread on the path_pop. In general I agree with the `poor_mans_string_paths` library now that I had to again write this myself. Before porting lets make sure this makes sense (in the other thread).